### ISSUE TITLE
Dropping python 3

### ISSUE THREAD

**Issue by hmaarrfk** [colinoflynn/pico-python#127]
Hey @colinoflynn 

do you think it is time to drop python2.7? most of the scientific stack is dropping it soon and it is kinda a blocker now for PR #126.

I really have no intention of working hard to ensure that 2.7 works, so I am in favour of dropping it.

Do any of our users need 2.7?

**Comment by colinoflynn** [comment]
I use 2.7 for a bunch of my work still in ChipWhisperer (based on various other requirements), but I can always just keep an old version around for that. PR #126 didn't fully break 2.7 though it seems, at least on other platforms?

**Comment by hmaarrfk** [comment]
Yeah, PR #126 didn't "break" any old functionaly that worked with python 2.7, but it definitely doesn't work on Python 2.7.

If 1/2 of the core devs still uses 2.7, then I think it makes sense to keep 2.7 around.

I mostly don't want to create busy work for future contributors. Which we've had a few :smile:

**Comment by hmaarrfk** [comment]
Anyway, leaving this here for reference: 
http://python3statement.org/
Many science projects are going to drop 2.7 support this year or next. Should be fun to see how this plays out.

**Comment by colinoflynn** [comment]
Yeah, it's coming :/ Mostly my problem w/ ChipWhipsperer is PySide, which will likely mean moving to PySide 2 or something else, which ends up being a lot. It may almost be a rewrite when it gets down to it (5 years of changes in architecture add up)... so I wouldn't say hold pico-python back for me! I hardly count as a core dev by commits anymore anyway ;-)

**Comment by hmaarrfk** [comment]
Take a look at [qtpy](https://github.com/spyder-ide/qtpy).

You *should* be able to write your code in a way that compatible between PySide/PySide2/PyQt4/PyQt5

**Comment by colinoflynn** [comment]
Will check it out - I think someone was actually working a little on that in one of the branches, but anything to make easier is good. Otherwise I might just open a hotdog stand instead and ignore Python 3 forever.
